pkg/kubelet/cm/memorymanager/policy_static_test.go

```diff
@@ -4216,3 +4646,975 @@ func Test_isAffinityViolatingNUMAAllocations(t *testing.T) {
+			expectedErr: nil,
+			features:    map[featuregate.Feature]bool{features.PodLevelResources: true, features.PodLevelResourceManagers: true},
+		},
+		{
+			name: "Failure: Pod-level == Guaranteed containers, has podSharedPool containers, PLR: Enabled, PLRM: Enabled",
+			pod: makeMultiContainerPodWithOptionsAndPodLevelResources(
+				"5Gi",
+				[]*containerOptions{},
+				[]*containerOptions{
+					{name: "c1", request: "2Gi", limit: "2Gi"},
+					{name: "c2", request: "3Gi", limit: "3Gi"},
+					{name: "c3"},
+				},
+			),
+			scope:       topologymanager.PodTopologyScope,
+			expectedErr: fmt.Errorf("pod rejected, sum of exclusive container memory requests equals pod budget, leaving no memory for shared containers"),
+			features:    map[featuregate.Feature]bool{features.PodLevelResources: true, features.PodLevelResourceManagers: true},
+		},
+		{
+			name: "Valid: Pod-level resources with standard and restartable init containers, PLR: Enabled, PLRM: Enabled",
+			pod: makeMultiContainerPodWithOptionsAndPodLevelResources(
+				"2Gi",
+				[]*containerOptions{
+					{name: "init-c1", request: "1Gi", limit: "1Gi"},
+					{name: "restartable-init-c1", request: "1Gi", limit: "1Gi", restartPolicy: &containerRestartPolicyAlways},
+				},
+				[]*containerOptions{
+					{name: "c1"},
+				},
+			),
+			scope:       topologymanager.PodTopologyScope,
+			expectedErr: nil,
+			features:    map[featuregate.Feature]bool{features.PodLevelResources: true, features.PodLevelResourceManagers: true},
+		},
+		{
+			name: "Valid: Pod-level resources equal init and standard container resources, PLR: Enabled, PLRM: Enabled",
+			pod: makeMultiContainerPodWithOptionsAndPodLevelResources(
+				"2Gi",
+				[]*containerOptions{
+					{name: "restartable-init-c1", request: "2Gi", limit: "2Gi", restartPolicy: &containerRestartPolicyAlways},
+				},
+				[]*containerOptions{
+					{name: "c1", request: "2Gi", limit: "2Gi"},
+				},
+			),
+			scope:       topologymanager.PodTopologyScope,
+			expectedErr: nil,
+			features:    map[featuregate.Feature]bool{features.PodLevelResources: true, features.PodLevelResourceManagers: true},
+		},
+		{
+			name: "Failure: Pod-level resources with shared standard init container and no available pool, PLR: Enabled, PLRM: Enabled",
+			pod: makeMultiContainerPodWithOptionsAndPodLevelResources(
+				"2Gi",
+				[]*containerOptions{
+					{name: "restartable-init-c1", request: "2Gi", limit: "2Gi", restartPolicy: &containerRestartPolicyAlways},
+					{name: "init-c1"},
+				},
+				[]*containerOptions{},
+			),
+			scope:       topologymanager.PodTopologyScope,
+			expectedErr: fmt.Errorf("pod rejected, pod has shared init containers but no memory available for them"),
+			features:    map[featuregate.Feature]bool{features.PodLevelResources: true, features.PodLevelResourceManagers: true},
+		},
+		{
+			name: "Valid: Pod-level resources with shared standard init container and no available pool, PLR: Enabled, PLRM: Enabled",
```

Comment:
The name should reflect why it si fine: `Valid: Pod-level resources with shared standard init container and following sidecar container exhausting the pool, no app container...` or something like that